deleteSwipe now emits the deleted message Id.

DeclineThyself opened the pull request:
Please change this to emit the mesId.
```javascript
export async function deleteSwipe(swipeId = null, mesId = chat.length - 1) {
//...
await eventSource.emit(event_types.MESSAGE_SWIPE_DELETED, { mesId, swipeId, newSwipeId });
```

https://github.com/SillyTavern/SillyTavern/pull/4573#issuecomment-3368843222

Wolfsblvt commented:
@codex review

Wolfsblvt commented on public/script.js:
```diff
@@ -7945,41 +7945,41 @@ export function hideSwipeButtons() {
  * @param {number?} swipeId - The ID of the swipe to delete. If not provided, the current swipe will be deleted.
  * @returns {Promise<number>|undefined} - The ID of the new swipe after deletion.
```
The method already has jsdoc, so if you add arguments, please update the jsdoc too. `mesId` should be listed here.

DeclineThyself replied:
I forgot, Thank you.

Wolfsblvt reviewed (commented):
Think this makes sense and works. Small adjustment for docs.

Wolfsblvt commented on public/script.js:
```diff
@@ -7945,41 +7945,41 @@ export function hideSwipeButtons() {
     }
 
     // Select the next swipe, or the one before if it was the last one
-    const newSwipeId = Math.min(swipeId, lastMessage.swipes.length - 1);
+    const newSwipeId = Math.min(swipeId, message.swipes.length - 1);
     syncSwipeToMes(null, newSwipeId);
```
Wow, codex correct here.
Please fix this. Should be as easy as passing the `mesId` into `syncSwipeToMes` as the first argument, instead of null.

DeclineThyself replied:
Fixed.

Commit:
deleteSwipe now emits the deleted message Id.

Wolfsblvt commented on public/script.js:
```diff
@@ -7942,44 +7942,45 @@ export function hideSwipeButtons() {
  *
- * @param {number?} swipeId - The ID of the swipe to delete. If not provided, the current swipe will be deleted.
+ * @param {number?} [swipeId = null] - The ID of the swipe to delete. If not provided, the current swipe will be deleted.
+ * @param {number?} [mesId = chat.length - 1] - The ID of the message to delete. If not provided, the last message will be deleted.
```
Well, we are not deleting the message, still the swipe. But that's very nitpicky now.
Just to be correct...

DeclineThyself replied:
No, not a nitpick.
Thank you.

Wolfsblvt replied:
It's just jsdoc. But yeah, if we are working on it, make it correct.

DeclineThyself commented:
Thank you for running codex. I completely missed that.

Wolfsblvt reviewed (approved):
Think this should be okay. Can't see anything off anymore.

Cohee1207 commented on public/script.js:
```diff
@@ -10486,7 +10487,7 @@ jQuery(async function () {
             const message = chat[this_edit_mes_id];
             const swipe_id = message.swipe_id;
-            await deleteSwipe(swipe_id);
+            await deleteSwipe(swipe_id, this_edit_mes_id);
```
`this_edit_mes_id` can be a string (don't ask). Either wrap it in `parseInt` or `Number` function

DeclineThyself replied:
Fixed: https://github.com/SillyTavern/SillyTavern/pull/4609/commits/1a21552bce5d782c62010a6c6c45e4e4844d0954

Cohee1207 commented on public/script.js:
```diff
@@ -7942,44 +7942,45 @@ export function hideSwipeButtons() {
  * @returns {Promise<number>|undefined} - The ID of the new swipe after deletion.
  */
-export async function deleteSwipe(swipeId = null) {
+export async function deleteSwipe(swipeId = null, mesId = chat.length - 1) {
```
I think it's better to use `messageId` in naming. Most of weirdly abbreviated params and variables come way back from TavernAI code.

DeclineThyself replied:
Fixed: https://github.com/SillyTavern/SillyTavern/pull/4609/commits/1a21552bce5d782c62010a6c6c45e4e4844d0954

Can this be added to a style guide? https://github.com/SillyTavern/SillyTavern/blob/staging/CONTRIBUTING.md

DeclineThyself replied:
Do you want this changed here too?
https://github.com/SillyTavern/SillyTavern/pull/4610

Cohee1207 replied:
> Can this be added to a style guide?

#4613 <= add that here

Commit:
Fixed:

this_edit_mes_id can be a string (don't ask). Either wrap it in parseInt or Number function
https://github.com/SillyTavern/SillyTavern/pull/4609#discussion_r2405534766

I think it's better to use messageId in naming. Most of weirdly abbreviated params and variables come way back from TavernAI code.
https://github.com/SillyTavern/SillyTavern/pull/4609#discussion_r2405537550

Cohee1207 committed:
Revert outside number conversions

Cohee1207 reviewed (approved):
Tested, works.
